usr/local/bin/grant_access

```
@@ -78,5 +120,14 @@ def get_args():
     return parser.parse_args()
 
 
+def valid_email(s):
+    user, _, domain = s.rpartition('@', 1)
+    return EMAIL_USER.match(user) and EMAIL_DOMAIN.match(domain)
+
+
+def parse_emails(emails):
+    return list(filter(valid_email, re.split(DELIMITERS, emails)))
```

Yeah, I thought about that. It will simply not grant access to users whose emails are mistyped. We could fail if any of a list of emails is invalid, but since you would just re-run the job to add the correct emails, we might as well grant access to the valid ones first and then you only have to re-run with the corrected emails and not all of them.